### PR TITLE
Add golangci-lint

### PR DESCRIPTION
**Pull request by rooneyshuman**
Adds a linter and makes updates to resolve linting errors

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,3 +16,6 @@ benchmark:
 checks:
 	go vet ${PKGS}
 	staticcheck ${PKGS}
+
```
Comment (inkel) [inline Makefile:19]: How about adding the linter as the last step of this target?
```suggestion
	make lint

```

File: Makefile [hunk 2]
```diff
@@ -16,3 +16,6 @@ benchmark:
...
+lint:
+	golangci-lint run -c .golangci.yml
```
Comment (inkel) [inline Makefile:21]: BTW, a "foolproof" way to run this even if it's not installed in the user system (although could it be a bit slow) is to replace this with:
```suggestion
	go run github.com/golangci/golangci-lint/cmd/golangci-lint@latest run -c .golangci.yml
```

File: Makefile (hunk 1 above)
Comment (rooneyshuman) [reply Makefile:19]: sounds good. I'll make the change

File: Makefile (hunk 2 above)
Comment (rooneyshuman) [reply Makefile:21]: I like it

**Review by logyball — APPROVED** [review]

**Pull request merged**